src/lib/WalletInfoCollector.ts

```diff
@@ -64,6 +69,7 @@ export default class WalletInfoCollector {
                 && !initialAccounts.some((account) => account.address === potentialVestingOwner.address)) {
                 // make sure a ledger vesting owner account get's added even if it has no balance or transaction history
                 initialAccounts.push(potentialVestingOwner);
+                hasHistoryOrBalance = true;
```

Comment:
This line should not depend on the condition `contracts.length > 0`.
As the second condition is not fulfilled for legacy accounts.

Also here one sees, that `hasHistoryOrBalance` is not a well chosen name.